File: check_onu_type.py
import socket, time

def cmd(s, c, w=3):
    s.send((c + '\r\n').encode())
    time.sleep(w)
    d = b''
    s.setblocking(False)
    try:
        while True:
            ch = s.recv(4096)
            if not ch:
                break
            d += ch
    except:
        pass
    s.setblocking(True)
    return d.decode('ascii', errors='ignore')

s = socket.socket()
s.settimeout(10)
s.connect(('136.1.1.100', 23))
time.sleep(1)
s.recv(4096)
s.send(b'zte\r\n')
time.sleep(1)
s.recv(4096)
s.send(b'zte\r\n')
time.sleep(2)
s.recv(4096)
cmd(s, 'terminal length 0', 1)

# Check ONU types for all ONUs on port 1/1/1
print('=== ONU AUTH INFO (types) ===')
print(cmd(s, 'show gpon onu baseinfo gpon-olt_1/1/1', 5))

# Check OMCI info for ONU 19 to see real model
print('=== ONU 19 OMCI VERSION ===')
print(cmd(s, 'show gpon remote-onu equip gpon-onu_1/1/1:19', 4))

# Compare with working ONU 1 
print('=== ONU 1 OMCI VERSION (working) ===')
print(cmd(s, 'show gpon remote-onu equip gpon-onu_1/1/1:1', 4))

# Check if config download is stuck
print('=== ONU 19 CONFIG DOWNLOAD ===')
print(cmd(s, 'show gpon onu config-download-status gpon-onu_1/1/1:19', 4))

s.close()

File: check_onu_type.php
<?php
// Check ONU type for 1/1/1:19 (ours) vs 1/1/4:2 (SmartOLT)
$fp = @fsockopen('136.1.1.100', 23, $errno, $errstr, 5);
if (!$fp) die("Telnet failed: $errstr\n");
stream_set_timeout($fp, 2);

function readUntil($fp, $timeout = 10) {
    $buf = '';
    $deadline = time() + $timeout;
    while (time() < $deadline) {
        $c = @fread($fp, 4096);
        if ($c === false || $c === '') { usleep(100000); continue; }
        $buf .= $c;
        if (preg_match('/[\w)][#>]\s*$/', $buf) || preg_match('/\]\s*:\s*$/', $buf)) break;
    }
    return $buf;
}

function waitFor($fp, $patterns, $timeout = 10) {
    $buf = '';
    $deadline = time() + $timeout;
    while (time() < $deadline) {
        $c = @fread($fp, 4096);
        if ($c === false || $c === '') { usleep(100000); continue; }
        $buf .= $c;
        foreach ($patterns as $p) {
            if (stripos($buf, $p) !== false) return $buf;
        }
    }
    return $buf;
}

function sendCmd($fp, $cmd) {
    fwrite($fp, "$cmd\r\n");
    usleep(500000);
    return readUntil($fp, 10);
}

waitFor($fp, ['Username:', 'login:']);
fwrite($fp, "zte\r\n");
waitFor($fp, ['Password:']);
fwrite($fp, "zte\r\n");
sleep(1);
readUntil($fp);

sendCmd($fp, "terminal length 0");

// Show ONU type for both
echo "=== ONU 1/1/1:19 (OURS) ===\n";
$out = sendCmd($fp, "show gpon onu detail-info gpon-onu_1/1/1:19");
echo $out . "\n\n";

echo "=== ONU 1/1/4:2 (SMARTOLT) ===\n";
$out = sendCmd($fp, "show gpon onu detail-info gpon-onu_1/1/4:2");
echo $out . "\n\n";

// Also check onu type in running-config  
echo "=== RUNNING-CONFIG OLT PORT 1/1/1 (show onu type) ===\n";
$out = sendCmd($fp, "show running-config interface gpon-olt_1/1/1");
echo $out . "\n\n";

echo "=== RUNNING-CONFIG OLT PORT 1/1/4 (show onu type) ===\n";
$out = sendCmd($fp, "show running-config interface gpon-olt_1/1/4");
echo $out . "\n\n";

fclose($fp);
